src/status/wallet/utils.nim

```diff
@@ -0,0 +1,74 @@
+proc confirmed*(self:PendingTransactionType):string =
+  result = "transaction:" & $self
+
+proc buildTokenTransaction(source, to, assetAddress: Address, value: float, transfer: var Transfer, contract: var Erc20Contract, gas = "", gasPrice = ""): EthSend =
+  contract = getErc20Contract(assetAddress)
+  if contract == nil:
+    raise newException(ValueError, fmt"Could not find ERC-20 contract with address '{assetAddress}' for the current network")
+  transfer = Transfer(to: to, value: eth2Wei(value, contract.decimals))
+  transactions.buildTokenTransaction(source, assetAddress, gas, gasPrice)
+
+proc sendTransaction*(source, to, value, gas, gasPrice, password: string, success: var bool, data = ""): string =
+  var tx = transactions.buildTransaction(
+    parseAddress(source),
+    eth2Wei(parseFloat(value), 18), gas, gasPrice, data
+  )
+
+  if to != "":
+    tx.to = parseAddress(to).some
+
+  result = eth.sendTransaction(tx, password, success)
+  if success:
+    trackPendingTransaction(result, $source, $to, PendingTransactionType.WalletTransfer, "")
+
+proc sendTokenTransaction*(source, to, assetAddress, value, gas, gasPrice, password: string, success: var bool): string =
+  var
+    transfer: Transfer
+    contract: Erc20Contract
+    tx = buildTokenTransaction(
+      parseAddress(source),
+      parseAddress(to),
+      parseAddress(assetAddress),
+      parseFloat(value),
+      transfer,
+      contract,
+      gas,
+      gasPrice
+    )
+
+  result = contract.methods["transfer"].send(tx, transfer, password, success)
+  if success:
+    trackPendingTransaction(result, $source, $to, PendingTransactionType.WalletTransfer, "")
+
+proc getEthBalance*(address: string): string =
+  var balance = getBalance(address)
+  result = hex2token(balance, 18)
+
+proc getGasPricePredictions*(): GasPricePrediction =
+  if status_settings.getCurrentNetwork() != Network.Mainnet:
+    # TODO: what about other chains like xdai?
+    return GasPricePrediction(safeLow: 1.0, standard: 2.0, fast: 3.0, fastest: 4.0)
+  let secureSSLContext = newContext()
+  let client = newHttpClient(sslContext = secureSSLContext)
+  try:
+    let url: string = fmt"https://etherchain.org/api/gasPriceOracle"
+    client.headers = newHttpHeaders({ "Content-Type": "application/json" })
+    let response = client.request(url)
+    result = Json.decode(response.body, GasPricePrediction)
+  except Exception as e:
+    echo "error getting gas price predictions"
+    echo e.msg
+  finally:
+    client.close()
+
+proc getTransfersByAddress*(address: string, toBlock: Uint256, limit: int, loadMore: bool): seq[Transaction] =
```

Comment:
I would not categorize the 4 below procs as utils. It is definitely a personal opinion so not blocking at all here. IMO utils function are not main function as those are

Reply:
@alaibe those procs are not logically belong to `WalletModel` from `status/wallet.nim` and I agree maybe `utils` is not the best name, maybe `wallet_helper` is better one, but my intention was to split and move those helper methods from the `status/wallet.nim` file.

Reply:
I would name the file proxy (sorry for not suggesting in the first place). but helper would work. I understood the intent and I agree with it. I was just a bit confused by what is utils

Reply:
@alaibe I updated this, you can proceed.